.github/workflows/nightly.yml

```diff
@@ -0,0 +1,143 @@
+name: Nightly
```

Comment:
is this file really that different from `push.yml`?

in `push.yml`, extra nightly steps can be conditional based on GH trigger action, see https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-github-context

Reply:
GitHub actions is disappointingly bad at sharing code. The code in `nightly` depends on a matrix for all changed branches. Theoretically the matrix build script could handle this but I don't want to convolute it any more than necessary.